main/views.py

```diff
@@ -1,3 +1,127 @@
+from django.contrib.auth import authenticate, login
+from django.contrib.auth.decorators import login_required
+from django.urls import reverse_lazy
+from django.views.generic import ListView, DetailView
+from django.views.generic.edit import FormView, CreateView, DeleteView
+
+from .forms import SiteToCheckForm, MyUserCreationForm
+from .models import SiteToCheck
+from .calculations import SiteDownChecker, modify_email
+
+
+def login_view(request):
+    username = request.POST['username']
+    password = request.POST['password']
+    user = authenticate(request, username=username, password=password)
+    if user is not None:
+        login(request, user)
+        # Redirect to a success page.
+        render(request, 'login.html')
+    else:
+        # Return an 'invalid login' error message.
+        render(request, 'login.html')
+
+
+class CreateNewUserView(CreateView):
+    template_name = 'register.html'
+    form_class = MyUserCreationForm
+
+    def form_valid(self, form):
+        user = form.save()
+        login(self.request, user)
+        return redirect('index')
+
+
+class IndexView(ListView):
+    template_name = 'index.html'
+    context_object_name = 'sites'
+
+    def get_context_data(self, **kwargs):
+        context = super(IndexView, self).get_context_data(**kwargs)
+        context['form'] = self.request.GET.get('form', SiteToCheckForm)
+        return context
+
+    def get_queryset(self):
+        return SiteToCheck.objects.filter(user=self.request.user)
+
+
+class AddSiteToCheckView(FormView):
+    template_name = 'index.html'
+    context_object_name = 'sites'
+    model = SiteToCheck
+    form_class = SiteToCheckForm
+
+    def form_valid(self, form):
+        cd = form.cleaned_data
+        if SiteToCheck.objects.filter(url=cd['url'], user=self.request.user).exists():
+            messages.error(self.request, 'The page already exists in database')
+        else:
+            data = SiteDownChecker(cd['url'], user=self.request.user).status()
+            success_message_text = f"The page hes been added. \n\nStatus: {data['last_status']}, Response time: \
+            {data['last_response_time']}"
+            messages.success(self.request, success_message_text)
+        return redirect('/')
+
+    def form_invalid(self, form):
+        messages.error(self.request, 'Please enter the correct URL')
+        return redirect('/')
+
+
+class SiteDetailView(DetailView):
+    model = SiteToCheck
+    template_name = 'details.html'
+    context_object_name = 'url'
+
+
+class SiteDeleteView(DeleteView):
+    model = SiteToCheck
+    success_url = reverse_lazy('index')
+
+    def get(self, *args, **kwargs):
+        message = f'{self.get_object().url} was deleted'
+        messages.success(self.request, message)
+        return self.post(*args, **kwargs)
+
+
+@login_required
+def update_email(request):
+    modify_email(request)
+    return redirect('/')
+
+
+@login_required
+def url_refresh(request, pk):
```

Comment:
To mógłby być też DetailView. Załatił by ci sprawę wyciągania obiektu i rzucania 404.